core/src/main/java/org/mule/runtime/core/routing/IdempotentMessageFilter.java

```diff
@@ -38,18 +41,15 @@
  * <b>EIP Reference:</b> <a href="http://www.eaipatterns.com/IdempotentReceiver.html">
  * http://www.eaipatterns.com/IdempotentReceiver.html</a>
  */
-public class IdempotentMessageFilter extends AbstractFilteringMessageProcessor implements Initialisable, Disposable {
+public class IdempotentMessageFilter extends AbstractInterceptingMessageProcessor implements Initialisable, Disposable {
```

Comment:
If we no longer implemented `process()` then we could clean up all of this significantly. But given we still implement `process()` this processor still needs to intercept the flow and therefore needs to control if next is invoked or not.  The only other way of control if next processor in the chain is invoked would be to return `null` or some special `Event` from this processor which the chain that uses to interrupt execution, but I'm not sure it's worth changing this, and not part of this PR for sure.